src/assets/VueOptionVsCompositionAPI/option-vs-composition-api.md

```diff
@@ -0,0 +1,186 @@
+   // data
+    const data = reactive({
+      name: 'Sagar Gurung',
+      age: 26,
+      votingStatus: "Not Eligible",
+      bornYear: 1997
+    })
+   ```
+
+2. ### Defining methods
+    Defining the methods are very simple. It is just as easy as simply defining functions in JavaScript.
+    ```js
+    // methods
+    function increaseAge() {
+      data.age++
+    }
+
+    function decreaseAge() {
+      data.age--
+    }
+   ```
+   The above function changes the `age` of the data defined in the `data` which is reactive.
+
+3. ### Defining computed
+   To define the computed property vue provides the method called `computed` through which we can explicitly define the computed function.
+    ```js
+   //computed
+    const getVotingStatus = computed(() => {
+      return (data.age >= 18) ? 'Eligible' : 'Not Eligible'
+    })
+   ```
+   Here the operation `getVotingStatus` returns the voting status based on the value of the data age.
+
+4. ### Defining watch
+    Similarly, vue has provided a `watch` function to watch the change in any data properties defined.
+    ```js
+   //watcher
+    watch(mainData.age, (oldAge, newAge) => {
+      if(oldAge > newAge) {
+        data.bornYear--
+      } else {
+        data.bornYear++
+      }
+    })
+   ```
+   We have watched data `age` in such a way that when age is increased the data `bornYear` will decrease and vice-versa.
+
+The overall script snippet code looks like this at last for the composition API:
+```js
+import { reactive, toRefs, computed , watch} from "vue";
+export default {
+  setup() {
+    // data
+    const data = reactive({
+      name: 'Sagar Gurung',
+      age: 26,
+      votingStatus: "Not Eligible",
+      bornYear: 1997
+    })
+    const mainData = toRefs(data)
+    // methods
+    function increaseAge() {
+      data.age++
+    }
+    function decreaseAge() {
+      data.age--
+    }
+    //computed
+    const getVotingStatus = computed(() => {
+      return (data.age >= 18) ? 'Eligible' : 'Not Eligible'
+    })
+    //watcher
+    watch(mainData.age, (oldAge, newAge) => {
+      if(oldAge > newAge) {
+        data.bornYear--
+      } else {
+        data.bornYear++
+      }
+    })
+
+    return {
+      ...mainData,
+      increaseAge,
+      decreaseAge,
+      getVotingStatus
+    }
+  }
+}
+```
+In the overall code, we have imported `reactive`, `toRefs`, `computed`, and `watch` needed for building components with Composition API. The `toRefs` is used to convert the data created with `reactive` to plain objects. Also, in Composition API we need to define all those `data`, `computed`, `watch`, and `methods` inside of it as done above.
+At last, all the things that have been defined `data`, `computed`, and `methods` needs to be returned to be exposed to the template.
+
+In this way, we can simply convert a component defined with `Option API` to `Composition API`. But this is just a basic example. There are lots of things that we can perform using composition API.
```

Comment:
```suggestion
In this demo application we have learnt the basics of how to convert a component defined with `Option API` to `Composition API`. However there are lots of other things that we can perform using the `Composition API`.
```

Reply:
As a reader I am interested in what are some of the other things the Composition API could do. Maybe list a few, give reference to the Vue.js documentation or point to another upcoming blog post...